src/StatusQ/Popups/StatusSearchPopup.qml

```diff
@@ -0,0 +1,192 @@
+    property string selectionBadgeSecondaryText: ""
+    property string selectionBadgeIdenticonColor: ""
+    property bool loading
+    property Component searchOptionsPopupMenu
+    function resetSelectionBadge() {
+        selectionBadgeImage = "";
+        selectionBadgePrimaryText = "Anywhere";
+        selectionBadgeSecondaryText = "";
+        selectionBadgeIdenticonColor = "";
+    }
+    property ListModel searchResults: ListModel {}
+
+    contentItem: Item {
+        width: parent.width
+        height: parent.height
+        ColumnLayout {
+            id: contentItemColumn
+            anchors.fill: parent
+            spacing: 0
+            Item {
+                Layout.fillWidth: true
+                Layout.preferredHeight: 63
+                StatusInput {
+                    id: inputText
+                    anchors.left: parent.left
+                    anchors.leftMargin: 8
+                    anchors.right: parent.right
+                    anchors.rightMargin: 8
+                    anchors.bottom: parent.bottom
+                    input.icon.name: "search"
+                    input.placeholderText: "Search"
+                    input.bgVisible: false
+                }
+            }
+            StatusMenuSeparator { Layout.fillWidth: true }
+            Item {
+                Layout.preferredWidth: parent.width
+                Layout.preferredHeight: 58
+                StatusButton {
+                    id: searchOptionsMenuButton
+                    anchors.left: parent.left
+                    anchors.leftMargin: 16
+                    anchors.verticalCenter: parent.verticalCenter
+                    implicitWidth: (contentItemRowLayout.width + 8)
+                    implicitHeight: 32
+
+                    contentItem: Item {
+                        anchors.fill: parent
+                        RowLayout {
+                            id: contentItemRowLayout
+                            anchors.centerIn: parent
+                            spacing: 2
+                            StatusBaseText {
+                                text: "In: "
+                            }
+                            StatusRoundedImage {
+                                implicitWidth: 16
+                                implicitHeight: 16
+                                visible: !!root.selectionBadgeImage
+                                image.source: root.selectionBadgeImage
+                            }
+                            StatusLetterIdenticon {
+                                implicitWidth: 16
+                                implicitHeight: 16
+                                letterSize: 11
+                                visible: ((root.selectionBadgeImage === "") && (root.selectionBadgePrimaryText !== "Anywhere"))
+                                color: root.selectionBadgeIdenticonColor
+                                name: root.selectionBadgePrimaryText
+                            }
```

Comment:
> We're using this mechanism where we first try StatusRoundedImage and fallback to LetterIdenticon if it fails (or go to letter identicon straight away if no image source was provided. Maybe we can use a similar logic here?

> Here's an example of this: https://github.com/status-im/StatusQ/blob/master/src/StatusQ/Components/StatusChatListItem.qml#L59-L103

Maybe this applies here as well?

Reply:
please see above

Reply:
Not really sure what you're referring to here...

Reply:
what do you mean?